p2p/src/peer_manager/mod.rs

```diff
@@ -790,7 +795,14 @@ where
         }
     }
 
-    fn handle_add_list_request(&mut self, peer: PeerId) {
+    fn handle_add_list_request(&mut self, peer_id: PeerId) {
+        let peer = self.peers.get_mut(&peer_id).expect("peer must be known");
+        if !peer.expect_addr_list_request {
+            log::debug!("Ignore unexpected address list request from peer {peer_id}");
+            return;
```

Comment:
No punishment?

Reply:
Yes, the same way it's done in Bitcoin Core. I don't see any risks with that.